tests/test_forward_ref.py

```diff
@@ -997,3 +997,26 @@ class allOfExpressions_(BaseModel):
     models = [allOfExpressions_, Expressions_]
     for m in models:
         m.model_rebuild(_types_namespace=types_namespace)
+
+
+def test_forward_ref_in_generic(create_module: Any) -> None:
+    """https://github.com/pydantic/pydantic/issues/6503"""
+
+    @create_module
+    def module():
+        import typing as tp
+
+        from pydantic import BaseModel
+
+        class Foo(BaseModel):
+            x: tp.Dict['tp.Type[Bar]', tp.Type['Bar']]
+
+        class Bar(BaseModel):
+            pass
+
+        Foo.model_rebuild()
+
```

Comment:
```suggestion
```
Isn't necessary due to automated rebuild, probably worth removing to make sure that doesn't stop working in the future.